### PR TITLE
Deprecate remaining cask commands.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

Depends on https://github.com/Homebrew/brew/pull/9222.

### CODE REVIEW

File: Library/Homebrew/cask/cmd.rb [hunk 1]
```diff
@@ -54,13 +54,23 @@ class Cmd
     }.freeze
 
     DEPRECATED_COMMANDS = {
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/cmd.rb:56]: Mind sticking a comment with `odeprecated` in it above here somewhere? That'll remind me to update these to `odisabled` for the minor release after next.

Comment (reitermarkus) [reply Library/Homebrew/cask/cmd.rb:56]: There is a call to `odeprecated` somehwere below where they are passed.

**Review by MikeMcQuaid — APPROVED** [review]

File: docs/Manpage.md [hunk 2]
```diff
@@ -71,39 +71,9 @@ Homebrew Cask provides a friendly CLI workflow for the administration of macOS a
...
-  <br>Creates the given *`cask`* and opens it in an editor.
-
-- `edit`
-  <br>Open the given *`cask`* for editing.
-
-- `fetch`
-  <br>Downloads remote application files to local cache.
-
 - `help`
   <br>Print help for `cask` commands.
```
Comment (MikeMcQuaid) [inline docs/Manpage.md:75]: @reitermarkus Can this get deprecated too? If so, I think we can remove this entire section

File: Library/Homebrew/cask/cmd.rb [hunk 3]
```diff
@@ -54,13 +54,23 @@ class Cmd
...
+      Cmd::List      => "brew list [--cask]",
+      Cmd::Outdated  => "brew outdated [--cask]",
+      Cmd::Reinstall => "brew reinstall [--cask]",
+      Cmd::Style     => "brew style",
+      Cmd::Uninstall => "brew uninstall [--cask]",
+      Cmd::Upgrade   => "brew upgrade [--cask]",
+      Cmd::Zap       => "brew upgrade --zap [--cask]",
     }.freeze
 
     sig { returns(String) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/cmd.rb:76]: I think the method below can probably be 🔥 entirely if `help` can also be deprecated.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for this @reitermarkus! Never thought I'd see the day where we'd be able to deprecate all these commands.

File: Library/Homebrew/cask/cmd.rb [hunk 4]
```diff
@@ -54,35 +54,26 @@ class Cmd
...
-      Cmd::Upgrade   => "brew upgrade --cask",
+      Cmd::Info      => "brew info [--cask]",
+      Cmd::Install   => "brew install [--cask]",
+      Cmd::List      => "brew list [--cask]",
+      Cmd::Outdated  => "brew outdated [--cask]",
+      Cmd::Reinstall => "brew reinstall [--cask]",
+      Cmd::Style     => "brew style",
+      Cmd::Uninstall => "brew uninstall [--cask]",
+      Cmd::Upgrade   => "brew upgrade [--cask]",
+      Cmd::Zap       => "brew upgrade --zap [--cask]",
```
Comment (ErikMinekus) [inline Library/Homebrew/cask/cmd.rb:74]: This should be `brew uninstall --zap [--cask]` right? There's no mention of `--zap` in `brew upgrade --help`.

Comment (MikeMcQuaid) [reply Library/Homebrew/cask/cmd.rb:74]: Yes, good catch, will fix.

Comment (MikeMcQuaid) [reply Library/Homebrew/cask/cmd.rb:74]: https://github.com/Homebrew/brew/pull/9298

**Pull request merged**